tests/ui/consts/const_in_pattern/issue-78057.stderr

```diff
@@ -5,13 +5,13 @@ LL |         FOO => {},
    |         ^^^
 
 error: unreachable pattern
-  --> $DIR/issue-78057.rs:14:9
+  --> $DIR/issue-78057.rs:12:9
    |
 LL |         FOO => {},
-   |         --- matches any value
+   |         ^^^ this pattern is irrefutable; subsequent arms are never executed
 LL |
 LL |         _ => {}
-   |         ^ unreachable pattern
+   |         ^^^^^^^ this arm is never executed
```

Comment:
this may get very verbose for longer arms, maybe keep pointing to the pattern?

Reply:
Looking at the other diagnostic changes I think we should just keep the previous `unreachable pattern`, considering the other message is so much more helpful now

Reply:
What do you mean with "the other message"? I'm afraid I'm not sure what you are proposing.

Reply:
Anyway: newbies tend to be more experienced with switch...case structures than pattern matching so I prefer saying "this arm is never executed" over "pattern is unreachable". It'd be kinda weird to say "this arm is never executed" and just point at the pattern rather than the pattern and the entire code block.

Reply:
I understand that argument, but it is orthogonal to the "make this diagnostic less spammy" change, so I would prefer to discuss it separately with wg-diagnostics, as it seems like a major divergence from existing diagnostics and we'd need to adjust other diagnostics, too.

Reply:
I'm in agreement with oli's points. We want to make the underlines as short as possible across the board. To address your concern around wording, you could say "this pattern cannot be reached, so its arm is never executed", while pointing only at the pattern.